docs/reference/query-dsl/query-string-query.asciidoc

```diff
@@ -144,8 +144,10 @@ When not explicitly specifying the field to search on in the query
 string syntax, the `index.query.default_field` will be used to derive
 which field to search on. If the `index.query.default_field` is not specified,
 the `query_string` will automatically attempt to determine the existing fields in the index's
-mapping that are queryable, and perform the search on those fields. Note that this will not
-include nested documents, use a nested query to search those documents.
+mapping that are queryable, and perform the search on those fields. 
+This will not include nested documents, use a nested query to search those documents.
+
+NOTE: For mappings with a large number of fields and _source disabled, searching across all queryable fields in the mapping could be expensive.
```

Comment:
I think its the "_all" field that when disabled is problematic. Since the "_all" field has been deprecated in 6 and is removed in 7, this is the default now if no "default field" is defined. We already have a Warning in the `default_field` parameter description above, please take a look if that covers what you are trying to say here, or if we should repeat or maybe move it here instead.

Reply:
Yeah _source is wrong, I mean't _all.
I feel like the warning might still be useful, we can drop the mention of the _all field since it's already deprecated though. The warning at the top is around how many fields might be searched without a default so that might help users think about why some fields are not being searched if they expect them to be, but I think it doesn't directly warn of possible performance impact even though that might be implied by searching lots of fields. Feels like it might be a bit of duplication but the second warning is both not inside a table (which might be more easily scanned over) and also is in a note panel to stand out a bit more. I don't feel super strongly that this needs to be included or maybe we can add a mention of performance impact if searching a great number of fields in the top section instead.

Reply:
Or just mention the limit is in place to prevent excessive performance impacts when searching so many fields, but impacts might still be felt even within the default limit.

Reply:
> I feel like the warning might still be useful

I agree, would you mind rephrasing it in this direction slightly and removing the mentioning of the `_source` field? I think this warning is good to have twice if its not a direct repetition of whats mentioned above.

Reply:
Ah, an probably wrap around ~80 characters, we try to keep the asciidoc files readable in smaller editor windows.

Reply:
Removed _source mention and wrapped at 80.
Let me know if this looks good to go or if you think it needs any more tweaking.